File: problem007/7.rb
def is_prime(n)
    return true if n == 2 or n == 1
    return false if (n % 2 == 0 and n != 2) or (n % 3 == 0 and n!= 3) or (n % 5 == 0 and n!= 5) or (n % 7 == 0 and n!= 7)
    2.upto(Math.sqrt(n).to_i){|x|
      return false if n % x == 0
    }
end


def nth_prime(n)
    pos, p = 0
    i = 2
    #look for prime numbers, starting at 2
    while pos < n do
      #increment pos every time we find a prime
      if is_prime(i) then
         p = i
         pos += 1
      end
      i += 1
    end
    return p
end

p nth_prime(10001)

File: problem007/7.js
#!/usr/bin/env node

function multipleOfFirstPrimes(n) {
    var checks = [2, 3, 5, 7].map(function(x) {
        return (n % x == 0 && n != x);
    });
    return checks.reduce(function(a, b) { return a || b}); 
}

function checkDivisible(n) {
    var limit = Math.sqrt(n);
    for (i = 2; i <= limit; i++) {
        if (n % i == 0) {
            return true;
        }
    }
    return false;
}

function isPrime(n) {
    return (n == 2 || n == 1) && !(multipleOfFirstPrimes(n)) || !checkDivisible(n);
}

function nthPrime(n) {
    var pos =  0;
    var candidate = 0;
    var i = 2;
    //look for prime numbers, starting at 2
    while (pos < n) {
        //increment pos every time we find a prime
        if (isPrime(i)) {
            candidate = i;
            pos++;
        }
        i++;
    }
    return candidate;
}

console.log(nthPrime(10001));
